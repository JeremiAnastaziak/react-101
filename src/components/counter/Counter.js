import React, { PureComponent } from 'react';

export default class Counter extends PureComponent {
  render() {
    const { decrement, value, increment } = this.props;
    return (
              <>
                <p>Redux counter</p>
                <button type="button" onClick={decrement}>-</button>
                <input type="text" value={value} />
                <button type="button" onClick={increment}>+</button>
              </>
    );
  }
}
