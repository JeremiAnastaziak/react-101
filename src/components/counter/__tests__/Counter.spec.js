import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../Counter';

describe('Counter component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Counter increment={() => {}} decrement={() => {}} value="1" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
