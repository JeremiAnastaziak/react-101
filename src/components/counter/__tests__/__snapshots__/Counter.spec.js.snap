// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component renders correctly 1`] = `
Array [
  <p>
    Redux counter
  </p>,
  <button
    onClick={[Function]}
    type="button"
  >
    -
  </button>,
  <input
    type="text"
    value="1"
  />,
  <button
    onClick={[Function]}
    type="button"
  >
    +
  </button>,
]
`;
